common/src/commonMain/kotlin/serialization/DurationSerializers.kt

```diff
@@ -45,6 +45,33 @@ public sealed class DurationAsLongSerializer(
     }
 }
 
+/** A [Duration] that is [serializable][Serializable] with [DurationInFloatingPointSecondsSerializer]. */
+public typealias DurationInFloatingPointSeconds = @Serializable(with = DurationInFloatingPointSecondsSerializer::class) Duration
+
+/** Serializer that encodes and decodes [Duration]s as a [Double] number of seconds */
+public object DurationInFloatingPointSecondsSerializer : KSerializer<Duration> {
+    override val descriptor: SerialDescriptor =
+        PrimitiveSerialDescriptor("dev.kord.common.serialization.DurationInFloatingPointSeconds", PrimitiveKind.DOUBLE)
+
+    override fun serialize(encoder: Encoder, value: Duration) {
+        when (val valueAsDouble = value.toDouble(MILLISECONDS)) {
+            Double.MIN_VALUE, Double.MAX_VALUE -> throw SerializationException(
```

Comment:
this check is not appropriate for `Duration.toDouble()` (it doesn't coerce but round instead), only for `Duration.toLong()`